### ISSUE TITLE
Unable to delete token starting with 'minus'

### ISSUE THREAD

**Issue by pkernevez** [ckan/ckan#7598]
## CKAN version
2.10.0

## Describe the bug
When trying to delete a token using the command line it fails when the token start with minus because it's interpreted like an option instead of the token.

### Steps to reproduce
Create tokens until you have one starting with '-xxxx'.
Like :
```
 ckan -c /srv/app/ckan.ini user token list "default"
Tokens([id] name - lastAccess):
	[-xC3ZcjRCxod9hJ6jpjpKoU2-xMnPdpaIAUuTyRz2HpRKfrZyq5fYNhbXlWjWSjKzFrxSgMTHOpZMLoW] datapusher - Never
	[3d0c7SpnrwrAgzPZf3RsmIPmK3h-optQJsdum4HEdZ4YoXGDQQx1tzEjvG8-Q4kmExYCTzwylkXovAYh] datapusher - Never
```
Then delete it:
```
# For the second one it's a success
 ckan -c /srv/app/ckan.ini user token revoke "3d0c7SpnrwrAgzPZf3RsmIPmK3h-optQJsdum4HEdZ4YoXGDQQx1tzEjvG8-Q4kmExYCTzwylkXovAYh"
API Token has been revoked

# For the first one it fails
ckan -c /srv/app/ckan.ini user token revoke "-xC3ZcjRCxod9hJ6jpjpKoU2-xMnPdpaIAUuTyRz2HpRKfrZyq5fYNhbXlWjWSjKzFrxSgMTHOpZMLoW"
Usage: ckan user token revoke [OPTIONS] ID
Try 'ckan user token revoke --help' for help.

Error: No such option: -x
```
Whatever the token is between string seperator (", ' or anything else), the token is interpreted like an option instead of a token.

### Expected behavior
Being able to delete all the generated tokens.

**Comment by pkernevez** [comment]
No, this is not working, the char '\' is interpreted as part of the key and the token is not found.
ckan -c /srv/app/ckan.ini user token revoke "\-xxxxx"
API Token not found
Aborted!

**Comment by ThrawnCA** [comment]
POSIX has syntax to handle this, which Click supports: Include a `--` by itself to indicate that options are finished and all further values should be literals. It's documented at https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02 and https://github.com/pallets/click/issues/249

So, you could write your revocation as:

    ckan -c /srv/app/ckan.ini user token revoke -- -xxxxx
